### PR TITLE
Fix per-env gravity randomization on Newton backend

### PR DESCRIPTION
**Pull request by ooctipus**
## Summary

`GRAVITY_VEC_W` on Newton's `ArticulationData`, `RigidObjectData`, and `RigidObjectCollectionData` was snapshotted from env 0 at construction, normalized to a unit direction, and broadcast to every env (and every body for the collection). Any in-place mutation of Newton's per-world `model.gravity` array — including the `randomize_physics_scene_gravity` event term, which writes per-env values via `model.gravity[env_ids] = ...` — was invisible to every consumer of `GRAVITY_VEC_W` and to the lazily-recomputed `projected_gravity_b`.

This PR binds `GRAVITY_VEC_W` directly to `SimulationManager.get_model().gravity` so it is a live view of the source of truth. The field now carries world-frame m/s² (per-env) instead of a unit direction; downstream consumers that need a unit projection normalize on read.

- New dedicated `projected_gravity_b_kernel` / `projected_gravity_b_2D_kernel` in `isaaclab_newton.assets.kernels` replace the generic `quat_apply_inverse_*` calls for gravity and normalize internally.
- `body_projected_gravity_b` in `isaaclab.envs.mdp.observations` normalizes the torch view (no-op for PhysX/OvPhysX, which still pre-normalize at construction).
- Warp kernels in `isaaclab_experimental` (`_projected_gravity_kernel`, `_flat_orientation_l2_kernel`) and humanoid task kernels (`_base_up_proj_kernel`, `_upright_posture_bonus_kernel`) switch from `gravity_w[0]` to `wp.normalize(gravity_w[i])` so per-env randomized gravity is respected.
- The parity-test mock provides a per-env `GRAVITY_VEC_W` shaped to match Newton's `model.gravity`.

PhysX and OvPhysX backends are unchanged: their `GRAVITY_VEC_W` remains a pre-normalized unit vector, and the added `F.normalize` on the torch path is a no-op there.

### Behavior change to note

`isaaclab_newton.assets.{ArticulationData,RigidObjectData,RigidObjectCollectionData}.GRAVITY_VEC_W` now exposes the raw m/s² gravity vector instead of a unit direction. Code that consumed it directly (rather than going through `projected_gravity_b`) needs to normalize before use. `RigidObjectCollectionData.GRAVITY_VEC_W` is now shaped `(num_instances, 3)` instead of `(num_instances, num_bodies, 3)`; `projected_gravity_b` retains its `(num_instances, num_bodies, 3)` shape via the dedicated 2D kernel.

## Test plan

Added `test_gravity_vec_w_tracks_model_gravity` to the three Newton asset test suites:

- `source/isaaclab_newton/test/assets/test_articulation.py`
- `source/isaaclab_newton/test/assets/test_rigid_object.py`
- `source/isaaclab_newton/test/assets/test_rigid_object_collection.py`

Each test pointer-equality-checks `GRAVITY_VEC_W.warp.ptr` against `SimulationManager.get_model().gravity.ptr`, mutates `model.gravity` per env in place, then asserts both `GRAVITY_VEC_W` and `projected_gravity_b` reflect the new values without `sim.step()` (avoiding orientation drift).

- [ ] `./isaaclab.sh -p -m pytest source/isaaclab_newton/test/assets/test_articulation.py::test_gravity_vec_w_tracks_model_gravity`
- [ ] `./isaaclab.sh -p -m pytest source/isaaclab_newton/test/assets/test_rigid_object.py::test_gravity_vec_w_tracks_model_gravity`
- [ ] `./isaaclab.sh -p -m pytest source/isaaclab_newton/test/assets/test_rigid_object_collection.py::test_gravity_vec_w_tracks_model_gravity`
- [ ] Sanity-check `randomize_physics_scene_gravity` is observable in a quick locomotion config (e.g. shadow_hand).
- [ ] Verify regression tests fail when the fix is reverted (per AGENTS.md).

### CODE REVIEW

File: source/isaaclab_experimental/test/envs/mdp/parity_helpers.py [hunk 1]
```diff
@@ -218,8 +218,11 @@ def __init__(self, num_envs=NUM_ENVS, num_joints=NUM_JOINTS, device=DEVICE, seed
         vel_np[:, 3:] = ang_vel_w_np
         self.root_com_vel_w = proxy_array(vel_np, dtype=wp.spatial_vectorf, device=device)
 
-        # Gravity direction constant (1D array to match kernel signatures)
-        self.GRAVITY_VEC_W = proxy_array([0.0, 0.0, -1.0], dtype=wp.vec3f, device=device)
+        # Gravity vector in world frame (m/s^2). Per-env shape ``(num_envs,)`` to
+        # match Newton's ``model.gravity``; warp kernels and the torch path
+        # normalize before projecting, so the magnitude is preserved here.
+        gravity_vec_w = np.tile(np.array([[0.0, 0.0, -9.81]], dtype=np.float32), (num_envs, 1))
```
Comment (StafaH) [inline source/isaaclab_experimental/test/envs/mdp/parity_helpers.py]: Better one-line: `gravity_vec_w = np.full((num_envs, 3), (0.0, 0.0, -9.81), dtype=np.float32)`

File: source/isaaclab_newton/isaaclab_newton/assets/articulation/articulation_data.py [hunk 2]
```diff
@@ -75,15 +74,10 @@ def __init__(self, root_view: ArticulationView, device: str):
...
 
-        # Convert to direction vector
-        gravity = wp.to_torch(SimulationManager.get_model().gravity)[0]
-        gravity_dir = torch.tensor((gravity[0], gravity[1], gravity[2]), device=self.device)
-        gravity_dir = normalize(gravity_dir.unsqueeze(0)).squeeze(0)
-        gravity_dir = gravity_dir.repeat(self._root_view.count, 1)
+        # Bind the per-env world-frame gravity vector directly to Newton's
+        # ``model.gravity`` array (shape ``(num_envs,)``, dtype ``vec3f``).
+        self.GRAVITY_VEC_W = ProxyArray(SimulationManager.get_model().gravity)
         forward_vec = torch.tensor((1.0, 0.0, 0.0), device=self.device).repeat(self._root_view.count, 1)
```
Comment (StafaH) [inline source/isaaclab_newton/isaaclab_newton/assets/articulation/articulation_data.py]: Can inline this here and elsewhere, to keep consistent, just use same numpy one-liner, isntead of making torch and then calling wp.from_torch

File: source/isaaclab_experimental/test/envs/mdp/parity_helpers.py [hunk 3]
```diff
@@ -218,8 +218,11 @@ def __init__(self, num_envs=NUM_ENVS, num_joints=NUM_JOINTS, device=DEVICE, seed
...
+        self.GRAVITY_VEC_W = proxy_array(gravity_vec_w, dtype=wp.vec3f, device=device)
```
Comment (StafaH) [inline source/isaaclab_experimental/test/envs/mdp/parity_helpers.py]: Let's use the official `ProxyArray(wp.array(...))`, this proxy_array is a helper that only exists here in experimental

File: source/isaaclab_newton/test/assets/test_rigid_object.py [hunk 4]
```diff
@@ -888,17 +888,17 @@ def test_gravity_vec_w(num_cubes, device, gravity_enabled):
...
+            expected_g = (0.0, 0.0, 0.0)
 
         # Play sim
         sim.reset()
 
         # Check that gravity is set correctly
-        assert cube_object.data.GRAVITY_VEC_W.torch[0, 0] == gravity_dir[0]
-        assert cube_object.data.GRAVITY_VEC_W.torch[0, 1] == gravity_dir[1]
-        assert cube_object.data.GRAVITY_VEC_W.torch[0, 2] == gravity_dir[2]
+        assert cube_object.data.GRAVITY_VEC_W.torch[0, 0] == expected_g[0]
```
Comment (StafaH) [inline source/isaaclab_newton/test/assets/test_rigid_object.py]: Can use torch.testing.all_close to compare arrays?

File: source/isaaclab_newton/test/assets/test_rigid_object_collection.py [hunk 5]
```diff
@@ -557,16 +557,18 @@ def test_gravity_vec_w(num_envs, num_cubes, device, gravity_enabled):
...
+        expected_g = (0.0, 0.0, -9.81) if gravity_enabled else (0.0, 0.0, 0.0)
 
         sim.reset()
 
         # Check if gravity vector is set correctly
         gravity_vec = object_collection.data.GRAVITY_VEC_W.torch
-        assert gravity_vec[0, 0, 0] == gravity_dir[0]
-        assert gravity_vec[0, 0, 1] == gravity_dir[1]
-        assert gravity_vec[0, 0, 2] == gravity_dir[2]
+        assert gravity_vec[0, 0] == expected_g[0]
```
Comment (StafaH) [inline source/isaaclab_newton/test/assets/test_rigid_object_collection.py]: torch.testing.all_close (or np.testing.all_close)

**Review by StafaH — COMMENTED** [review]
Thanks @ooctipus, left some minor comments.

**Review by StafaH — APPROVED** [review]
